parlai/core/torch_agent.py

```diff
@@ -194,7 +199,7 @@ def map_valid(self, obs_batch, sort=True, is_valid=lambda obs: 'text_vec' in obs
                     padded_ys[i, :y.shape[0]] = y
             ys = padded_ys
 
-        return Batch(xs, ys, labels, valid_inds)
+        return Batch(xs, torch.LongTensor(x_lens), ys, labels, valid_inds)
```

Comment:
I'd prefer this to be put into a longtensor earlier in the function

Reply:
Fixed